<launch>
  <!-- Pose Adapter - 电表巡检主节点 -->
  
  <!-- 相机参数 -->
  <arg name="camera_width" default="1280"/>
  <arg name="camera_height" default="720"/>

  <!-- Go2 网络接口（Unitree SDK 使用的网卡，例如 eth0） -->
  <arg name="network_interface" default="eth0"/>
  
  <!-- 电表尺寸（米） -->
  <arg name="meter_width" default="0.1"/>
  <arg name="meter_height" default="0.15"/>
  
  <!-- 控制参数 -->
  <arg name="target_distance" default="1"/>
  <arg name="target_ratio" default="0.5"/>
  
  <!-- 模型路径 -->
  <arg name="yolo_model_path" default="models/yolov8n.pt"/>
  <arg name="use_paddle_ocr" default="true"/>
  
  <!-- 标定文件（留空则使用节点内默认内参） -->
  <arg name="calib_file" default="$(find pose_adapter)/config/default_camera_calib.yaml"/>
  
  <!-- Pose Adapter 主节点 -->
  <node name="pose_adapter" pkg="pose_adapter" type="pose_adapter_node.py" output="screen">
    <param name="camera_width" value="$(arg camera_width)"/>
    <param name="camera_height" value="$(arg camera_height)"/>
    <param name="meter_width" value="$(arg meter_width)"/>
    <param name="meter_height" value="$(arg meter_height)"/>
    <param name="target_distance" value="$(arg target_distance)"/>
    <param name="target_ratio" value="$(arg target_ratio)"/>
    <param name="yolo_model_path" value="$(arg yolo_model_path)"/>
    <param name="use_paddle_ocr" value="$(arg use_paddle_ocr)"/>
    <param name="calib_file" value="$(arg calib_file)"/>
    <param name="network_interface" value="$(arg network_interface)"/>
  </node>
  
  <!-- 图像查看（可选） -->
  <arg name="show_debug_image" default="true"/>
  <group if="$(arg show_debug_image)">
    <node name="image_view" pkg="image_view" type="image_view" output="screen">
      <remap from="image" to="/pose_adapter/debug_image"/>
    </node>
  </group>
  
</launch>